README.md

```diff
@@ -1,33 +1,126 @@
+
+1. Make sure you have Homebrew installed.
+   * Instructions: https://brew.sh/
+2. Install `coreutils` (needed for tools e.g., `sha256sum`) and `jq`:
+   ```bash
+   brew install coreutils jq
+   ```
+3. Ensure the newly installed tools are added to your `PATH`:
+   ```bash
+   echo 'export PATH="$PATH:/opt/homebrew/bin/:/usr/local/opt/coreutils/libexec/gnubin"' >> "${HOME}/.bashrc"
+   ```
+   Above, we rely on `.bashrc`, as the main commands from this repository are to be executed via Bash.
+4. Clone this repository: 
+   ```
+   git clone git@github.com:dfinity/sns-testing.git
+   cd sns-testing
+   ```
+5. Clone the dependency repositories and run the installation script:
+   ```bash
+   git clone https://github.com/dfinity/internet-identity.git
+   git clone https://github.com/dfinity/nns-dapp.git
+   bash install.sh
+   ```
+6. Start a local replica (this will keep running in the current console; press ⌘+C to stop):
+   ```bash
+   DFX_NET_JSON="${HOME}/.config/dfx/networks.json"
+   mkdir -p "$(dirname "${DFX_NET_JSON}")"
+   cp "$DFX_NET_JSON" "${DFX_NET_JSON}.tmp" 2>/dev/null  # save original config if present
+   echo '{
+      "local": {
+         "bind": "0.0.0.0:8080",
+         "type": "ephemeral",
+         "replica": {
+            "subnet_type": "system"
+         }
+      }
+   }' > "${DFX_NET_JSON}"
+   ./bin/dfx start --clean; \
+   mv "${DFX_NET_JSON}.tmp" "$DFX_NET_JSON" 2>/dev/null  # restore original config if it was present
+   ```
+
+   This should print the dashboard URL, e.g.:
+
+    ```
+    Awaiting local replica ...
+    Dashboard: http://localhost:35727/_/dashboard
+    ```
+
+7. Open another Bash console:
+   ```bash
+   bash
+   ```
+   and run the setup script:
+   ```bash
+   bash-3.2$ ./setup_locally.sh
+   ```
+   After this step, you can also access the [NNS frontend dapp](http://qsgjb-riaaa-aaaaa-aaaga-cai.localhost:8080/) from the browser.
+
+
+8. To validate the testing environment, run the example dapp shipped with this repository through the entire SNS lifecycle:
+   ```bash
+   bash-3.2$ ./run_basic_scenario.sh
+   ```
+   If the basic scenario finished successfully, you should see the message
+    `Basic scenario has successfully finished.` on the last line of the output.
+
+   Observe the newly created SNS instance via the [NNS frontend dapp](http://qsgjb-riaaa-aaaaa-aaaga-cai.localhost:8080/). When you try to login for the first time, you will need to register a new Internet Identity for testing.
+
+   > If you have successfully executed the above commands, you are now ready to [test your own dapp's SNS decentralization](#lifecycle).
+
+9. Clean-up (after you are done testing):
+    ```bash
+    bash-3.2$ ./cleanup.sh
+    ```
+    It should now be possible to repeat the scenario starting from step 1.
```

Comment:
Installing homebrew again should not be needed.

Reply:
Good catch. I've merged the first two steps and factored in out into step 0, s.t. this sentence now makes sense. :-)

Reply:
Thank you!